pkg/resource/deploy/step.go

```diff
@@ -385,11 +386,20 @@ type deleteProtectedError struct {
 }
 
 func (d deleteProtectedError) Error() string {
-	return fmt.Sprintf("resource %[1]q cannot be deleted\n"+
+	return d.err(runtime.GOOS)
+}
+
+func (d deleteProtectedError) err(os string) string {
+	msg := fmt.Sprintf("resource %[1]q cannot be deleted\n"+
 		"because it is protected. To unprotect the resource, "+
 		"either remove the `protect` flag from the resource in your Pulumi "+
 		"program and run `pulumi up`, or use the command:\n"+
 		"`pulumi state unprotect '%[1]s'`", d.urn)
```

Comment:
There's also a load of quotes like this in the config commands that we show in help, one in preview that we show for --plan. We should probably build this solution such that it can apply to those texts as well. Maybe a global var for the shell quote char, or a `Quote(string) string` method.